### DISCUSSION TITLE
model.metabolites.M_thf_LSQBKTc_RSQBKT.summary()

### DISCUSSION THREAD

**Discussion by Rahmaabubakr** [opencobra/cobrapy#1359] [Q&A] [answered]
Hello COBRApy Community,

When i used this command 

model.metabolites.M_thf_LSQBKTc_RSQBKT.summary()

I got this error
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File [~/RA/lib/python3.10/site-packages/cobra/core/dictlist.py:525](https://file+.vscode-resource.vscode-cdn.net/home/rahma/RA/~/RA/lib/python3.10/site-packages/cobra/core/dictlist.py:525), in DictList.__getattr__(self, attr)
    524 try:
--> 525     return DictList.get_by_id(self, attr)
    526 except KeyError:

File [~/RA/lib/python3.10/site-packages/cobra/core/dictlist.py:73](https://file+.vscode-resource.vscode-cdn.net/home/rahma/RA/~/RA/lib/python3.10/site-packages/cobra/core/dictlist.py:73), in DictList.get_by_id(self, id)
     72 """Return the element with a matching id."""
---> 73 return list.__getitem__(self, self._dict[id])

KeyError: 'M_thf_LSQBKTc_RSQBKT'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
[/home/rahma/RA/habolla.ipynb](https://file+.vscode-resource.vscode-cdn.net/home/rahma/RA/habolla.ipynb) Cell 120 line 1
----> [1](vscode-notebook-cell:/home/rahma/RA/habolla.ipynb#Y233sZmlsZQ%3D%3D?line=0) model.metabolites.M_thf_LSQBKTc_RSQBKT.summary()

File [~/RA/lib/python3.10/site-packages/cobra/core/dictlist.py:527](https://file+.vscode-resource.vscode-cdn.net/home/rahma/RA/~/RA/lib/python3.10/site-packages/cobra/core/dictlist.py:527), in DictList.__getattr__(self, attr)
    525     return DictList.get_by_id(self, attr)
    526 except KeyError:
--> 527     raise AttributeError(f"DictList has no attribute or entry {attr}")

AttributeError: DictList has no attribute or entry M_thf_LSQBKTc_RSQBKT

**Comment by cdiener** [comment, marked as answer]
That just means there is no metabolite with that name in the model. You can print a list of all metabolites with `print(list(models.metabolites))`.

**Reply by Midnighter** [reply]
Also worth mentioning, that you can [query metabolites](https://cobrapy.readthedocs.io/en/latest/autoapi/cobra/core/dictlist/index.html#cobra.core.dictlist.DictList.query) by name or other attributes.
